presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -351,6 +356,92 @@ private PlanBuilder project(PlanBuilder subPlan, Iterable<Expression> expression
                 analysis.getParameters());
     }
 
+    private PlanBuilder project(
+            PlanBuilder subPlan,
+            List<FunctionCall> aggregates,
+            ImmutableList<Expression> groupAndSortArguments,
+            List<Expression> argumentsAfterFilter,
+            List<FunctionCallWithFinalArguments> functionCallToFinalArguments)
+    {
+        TranslationMap translations = new TranslationMap(subPlan.getRelationPlan(), analysis, lambdaDeclarationToSymbolMap);
```

Comment:
I would divide this method into two. First I would handle the projection related to filter predicates and then projection with arguments. The way you wrote it is very mixed up, hard to follow the code.